kinesis/kinesis.go

```diff
@@ -562,22 +563,42 @@ func (outputPlugin *OutputPlugin) randomString() string {
 	return string(outputPlugin.random.buffer)
 }
 
+func getFromMap(dataKey string, record map[interface{}]interface{}) interface{} {
+	for k, v := range record {
+		currentKey := stringOrByteArray(k)
+		if currentKey == dataKey {
+			return v
+		}
+	}
+
+	return ""
+}
+
 // getPartitionKey returns the value for a given valid key
 // if the given key is empty or invalid, it returns a random string
 func (outputPlugin *OutputPlugin) getPartitionKey(record map[interface{}]interface{}) string {
 	partitionKey := outputPlugin.partitionKey
 	if partitionKey != "" {
-		for k, v := range record {
-			dataKey := stringOrByteArray(k)
-			if dataKey == partitionKey {
-				value := stringOrByteArray(v)
+		partitionKeys := strings.Split(partitionKey, "->")
+		num := len(partitionKeys)
+		for count, dataKey := range partitionKeys {
+			newRecord := getFromMap(dataKey, record)
+			if count == num-1 {
+				value := stringOrByteArray(newRecord)
 				if value != "" {
 					if len(value) > partitionKeyMaxLength {
 						value = value[0:partitionKeyMaxLength]
 					}
 					return value
 				}
 			}
+			_, ok := newRecord.(map[interface{}]interface{})
+			if ok {
+				record = newRecord.(map[interface{}]interface{})
+			} else {
+				logrus.Errorf("[kinesis %d] The partition key has a bad format; Will return a random string for it", outputPlugin.PluginID)
```

Comment:
Change to "The partition key could not be found in the record, using a random string instead"

Reply:
Updated the words